zephyr/lib/userspace_helper.c

```diff
@@ -156,6 +156,47 @@ int user_memory_init_shared(k_tid_t thread_id, struct processing_module *mod)
 	return k_mem_domain_add_thread(comp_dom, thread_id);
 }
 
+int user_add_memory(struct k_mem_domain *domain, uintptr_t addr, size_t size, uint32_t attr)
```

Comment:
it probably better if we fail if size or addr are not page aligned, that way we force the client users to think about whats in the domain (rather than silently growing the domain to include the whole page).